src/periphery/MarketLens.sol

```diff
@@ -0,0 +1,76 @@
+
+import "interfaces/IBentoBoxV1.sol";
+import "interfaces/ICauldronV2.sol";
+import {IERC20, IStrictERC20} from "BoringSolidity/interfaces/IERC20.sol";
+import "utils/CauldronLib.sol";
+
+contract MarketLens {
+    address payable public owner;
+    uint8 constant DECIMALS = 18;
+
+    constructor() {
+        owner = payable(msg.sender);
+    }
+
+    function getInterestPerYear(ICauldronV2 cauldron) external view returns (uint64 interestPerYear) {
+        (, , uint64 interestPerSecond) = cauldron.accrueInfo();
+        interestPerYear = CauldronLib.getInterestPerYearFromInterestPerSecond(interestPerSecond);
+    }
+
+    function getMaxBorrowForCauldronV2(ICauldronV2 cauldron) external view returns (uint256 maxBorrow) {
+        IBentoBoxV1 bentoBox = IBentoBoxV1(cauldron.bentoBox());
+        IERC20 mim = IERC20(cauldron.magicInternetMoney());
+        uint256 poolBalance = bentoBox.balanceOf(mim, address(cauldron));
+        maxBorrow = bentoBox.toAmount(mim, poolBalance, false);
+    }
+
+    function getMaxBorrowForCauldronV3(ICauldronV3 cauldron) external returns (uint256 maxBorrow) {
+        IBentoBoxV1 bentoBox = IBentoBoxV1(cauldron.bentoBox());
+        IERC20 mim = IERC20(cauldron.magicInternetMoney());
+        uint256 poolBalance = bentoBox.balanceOf(mim, address(cauldron));
+        uint256 mimInBentoBox = bentoBox.toAmount(mim, poolBalance, false);
+        uint256 userBorrowLimit = getUserBorrowLimit(cauldron);
+        maxBorrow = min(userBorrowLimit, mimInBentoBox);
+    }
+
+    // Why does this modify state? Want to be able to use view but cannot.
+    function getUserBorrowLimit(ICauldronV3 cauldron) public returns (uint256 userBorrowLimit) {
+        (uint256 totalLimit, uint256 borrowPartPerAddress) = cauldron.borrowLimit();
+        userBorrowLimit = min(totalLimit, borrowPartPerAddress);
+    }
+
+    function getTotalMimBorrowed(ICauldronV2 cauldron) external view returns (uint256 totalMimBorrowed) {
+        Rebase memory totalBorrow = cauldron.totalBorrow();
+        totalMimBorrowed = totalBorrow.elastic;
+    }
+
+    function getTvl(ICauldronV2 cauldron) external view returns (uint256 tvl) {
+        IBentoBoxV1 bentoBox = IBentoBoxV1(cauldron.bentoBox());
+        uint256 totalTokensDeposited = bentoBox.toAmount(cauldron.collateral(), cauldron.totalCollateralShare(), false);
+        tvl = (totalTokensDeposited * 10 ** DECIMALS) / getOracleExchangeRate(cauldron);
+    }
+
+    function getOracleExchangeRate(ICauldronV2 cauldron) public view returns (uint256 exchangeRate) {
+        IOracle oracle = IOracle(cauldron.oracle());
+        bytes memory oracleData = cauldron.oracleData();
+        exchangeRate = oracle.peekSpot(oracleData);
+    }
+
+    // How can I get the LIQUIDATION_MULTIPLIER constant?
+    function getLiquidationFee(ICauldronV2 cauldron) public view {
+        // liquidationFee = cauldron.LIQUIDATION_MULTIPLIER - 100_000;
+    }
+
+    // Where should I put these utility functions?
```

Comment:
Suggestions on where to move these to?

Reply:
CauldronLib.sol no? as long as it concerns Cauldron it's fine with me, if we feel like it need more separation later we could have CauldronLiquidationLib.sol

Reply:
Moved it to src/libraries/MathLib.sol